pkg/console/subresource/route/route.go

```diff
@@ -162,13 +162,36 @@ func wildcard() routev1.WildcardPolicyType {
 	return routev1.WildcardPolicyNone
 }
 
+// Use the best host option for the route.
+// Initially this will be Route.Spec.Host, but once an ingress has
+// been admitted, it should be the host value from the ingress.
+func GetBestHost(route *routev1.Route) string {
```

Comment:
I think we should have tests these functions

Reply:
agree

Reply:
I'll write tests once we are happy with impl.  I just revised all of these funcs per some discussions to ensure we got it right.  I think the new impl is better.